src/main/java/uk/gov/hmcts/probate/service/filebuilder/IronMountainFileService.java

```diff
@@ -139,15 +131,19 @@ private SolsAddress getAddress(CaseData caseData, int granteeNumber) {
     }
 
     private SolsAddress getAdditionalExecutorAddress(CaseData caseData, int index) {
-        if (caseData.getAdditionalExecutorsApplying().size() >= (index + 1)) {
-            return caseData.getAdditionalExecutorsApplying().get(index).getValue().getApplyingExecutorAddress();
+        if (caseData.getAdditionalExecutorsApplying() != null) {
+            if (caseData.getAdditionalExecutorsApplying().size() >= (index + 1)) {
+                return caseData.getAdditionalExecutorsApplying().get(index).getValue().getApplyingExecutorAddress();
+            }
         }
         return EMPTY_ADDRESS;
     }
 
     private String getApplyingExecutorName(CaseData caseData, int index) {
-        if (caseData.getAdditionalExecutorsApplying().size() >= (index + 1)) {
-            return caseData.getAdditionalExecutorsApplying().get(index).getValue().getApplyingExecutorName();
+        if (caseData.getAdditionalExecutorsApplying() != null) {
```

Comment:
You don't need a nested if statement here, you can do it in one:
if ((caseData.getAdditionalExecutorsApplying()) != null && (caseData.getAdditionalExecutorsApplying().size() >= (index + 1))) - same for the lines below

Reply:
👍